test/gtest/intervals.cpp

```diff
@@ -0,0 +1,95 @@
+#include "support/intervals.h"
+#include "ir/utils.h"
+#include "gtest/gtest.h"
+
+using namespace wasm;
+
+TEST(IntervalsTest, TestEmpty) {
+  std::vector<Interval> intervals;
+  std::vector<int> expected;
+  ASSERT_EQ(IntervalProcessor::filterOverlaps(intervals), expected);
+}
+
+TEST(IntervalsTest, TestOne) {
+  std::vector<Interval> intervals;
+  intervals.emplace_back(Interval{0, 2, 5});
```

Comment:
Since we're using emplace_back, we can just pass the arguments to the constructor without having to construct and then move a temporary object.
```suggestion
  intervals.emplace_back(0, 2, 5);
```